### ISSUE TITLE
[Bugfix 2.0] LCD problem introduced by commit "Move number-to-string functions to libs"

### ISSUE THREAD

**Issue by Bergerac56** [MarlinFirmware/Marlin#14276]
A bug seems to have been introduced by commit 2a96d4e23ad0090dc371710d079af731136dfe97
Previous versions were working fine.
When choosing to print a file on a graphic LCD, the first line is screwed up. Here what it gives:
![mini-IMG_20190612_115210](https://user-images.githubusercontent.com/27892880/59342619-68dcc900-8d0a-11e9-95fd-6a62d38ee7dc.jpg)

**Comment by shitcreek** [comment]
Confirmed bug. I've got the man, @thinkyhead, on it.

**Comment by tpruvot** [comment]
unsure if its related to this commit... There is no numbers printed on this screen in theory... 

btw wasn't "Print" supposed to be the default choice on this "question" ?

But the bug seems present for all... same on my F1 board with dogm TFT

**Comment by LinFor** [comment]
I have same problem with last-days builds of Marlin.

**Comment by tpruvot** [comment]
kinda weird bug... This change almost fixes it, but we lose the "p" of print

```
--- a/Marlin/src/lcd/menu/menu_sdcard.cpp
+++ b/Marlin/src/lcd/menu/menu_sdcard.cpp
@@ -82,10 +82,11 @@ inline void sdcard_start_selected_file() {
 #if ENABLED(SD_MENU_CONFIRM_START)

   void menu_sd_confirm() {
+    const char* msg = _UxGT("Start print");
     do_select_screen(
       PSTR(MSG_BUTTON_PRINT), PSTR(MSG_BUTTON_CANCEL),
       sdcard_start_selected_file, ui.goto_previous_screen,
-      PSTR(MSG_START_PRINT " "), card.longest_filename(), PSTR("?")
+      msg, card.longest_filename(), PSTR("?")
     );
   }
```

**Comment by Bergerac56** [comment]
Just tested your patch. Strange this "missing p" ! 
It seems that the "p" is replaced by the "_" between "start" and "print"

edit:
I made some tests with other messages for "Start print". Everything is correctly displayed provided there is no " " (space) between characters: "StartPrint" is correctly printed but "Start   print" is not. 
You can add as many "space" as you want, they will not be displayed and just mask the first character of the following word by placing one space (and only one).

I suspect that the space (ascii 32) is not correctly processed and introduces this "p" bug

**Comment by tpruvot** [comment]
interesting, maybe there is a kind of "nbsp;" char available..

**Comment by tpruvot** [comment]
fixed with #14292

**Comment by Bergerac56** [comment]
The workaround works fine. Till now, I did not find any side effect somewhere else.
